ocs_ci/ocs/platform_nodes.py

```diff
@@ -3459,3 +3470,109 @@ def wait_for_nodes_to_stop_or_terminate(self, nodes):
+                "Only RHCOS is supported for ROSA HCP. Check up "
+                "https://docs.redhat.com/en/documentation/red_hat_openshift_service_on_aws/4/html/"
+                "tutorials/getting-started-with-rosa#underlying-node-operating-system"
+            )
+        if num_nodes is None:
+            raise ValueError("num_nodes must be provided")
+
+        node_conf = node_conf or {}
+
+        node_list = self.create_nodes(node_conf, node_type, num_nodes)
+        # attach nodes will be skipped due to AWSNodes logic for non-"upi" deployment
+        self.attach_nodes_to_cluster(node_list)
+
+    def create_nodes(self, node_conf, node_type, num_nodes):
+        """
+        Create nodes by creating or updating a machine pool.
+
+        Args:
+            node_conf (dict): Node configuration for machine pool.
+            node_type (str): Type of node to be created.
+            num_nodes (int): Number of node instances to create.
+
+        Returns:
+            list: List of AWSNode objects (no known use case for this for non UPI deploymen, which is ROSA HCP,
+            implemented due to common create_nodes signature)
+        """
+        node_list = []
+        cluster_name = config.ENV_DATA.get("cluster_name")
+        node_conf = NodeConf(**node_conf)
+        machine_pool_id = node_conf.get("machinepool")
+        machine_pools = MachinePools(cluster_name=cluster_name)
+        machine_pool = machine_pools.filter(id=machine_pool_id, pick_first=True)
+
+        if (
+            machine_pool.exist
+            and node_conf.get("instance_type")
+            and machine_pool.instance_type != node_conf.get("instance_type")
+        ):
+            raise UnavailableResourceException(
+                f"MachinePool '{machine_pool_id}' "
+                "found with different instance type. "
+                "The test brakes logic, aborting test."
+            )
+        elif machine_pool.exist and machine_pool.instance_type == node_conf.get(
+            "instance_type"
+        ):
+            logger.info(f"MachinePool '{machine_pool_id}' found. Updating MachinePool")
+            node_conf["replicas"] = machine_pool.replicas + num_nodes
+            machine_pools.edit_machine_pool(node_conf, wait_ready=True)
+        elif not machine_pool.exist:
+            logger.info(
+                f"MachinePool '{machine_pool_id}' not found. Creating new MachinePool"
+            )
+            # create random machinepool name if not provided
+            node_conf["machinepool"] = machine_pool_id or "mp_" + random_string(3)
+            node_conf["instance_type"] = (
+                node_conf.get("instance_type")
+                or config.ENV_DATA["worker_instance_type"]
+            )
+            machine_pools.create_machine_pool(node_conf, num_nodes)
+
+        node_conf["zone"] = self.az.get_zone_number()
+        for _ in range(num_nodes):
+            node_list.append(ROSAHCPNode(node_conf, constants.RHCOS))
```

Comment:
Can you add a comment or a log message for this code? It needs to be clarified.

Reply:
sure, added 
```
# adding created nodes to the list of a nodes returned by this method
```

Reply:
Yes, sure.